### ISSUE TITLE
Support other extensions (e.g. jsx)

### ISSUE THREAD

**Issue by bruderstein** [jballant/CommonJSAutoComplete#4]
It would be nice if it supported alternative extensions, such as jsx.

Currently, if `components/SomeComponent.jsx` exists, it will add the following after the "var"

```
var SomeComponent = require('./components/SomeComponent')
```

For .jsx, you need the .jsx in the require (I believe there's a way to make require find this, depending on environment, but it would be nice to have the option to add the extension for "non-standard" extensions).  Perhaps a list in the config of supported extensions, and a list of extensions that aren't needed for require?

E.g. you have configured require to load .jsx, but .coffee (!) is automatically transpiled, and you need to specify it.

```
./SomeComponent.jsx
./controller.coffee
////////////////

var SomeComponent = require('./SomeComponent')
var controller = require('./controller.coffee')

```

Configured:     

```
Supported Extensions:  js, jsx, coffee
Extension required for: coffee 
```

**Comment by jballant** [comment]
I apologize, but I'm not sure I fully understand the issue.

I believe that you shouldn't need the extension for these transpiled files. For example, "jsx" should be registered as an extension in node with a module like [node-jsx](https://www.npmjs.com/package/node-jsx) and requiring the file without the extension should work just fine. Whatever transpiles your code should register the extension, or you could always register it manually.

In terms of client-side code, transpilers like [browserify](https://www.npmjs.com/package/browserify) or [Webpack](https://www.npmjs.com/package/webpack) let you specify supported extensions.

We use this on our React project currently, so we know that it works.

Proper support for CoffeeScript (and possibly TypeScript) is something I would like to add, but that is a separate problem from the extension.

Basically, I'm not sure what the use your feature other than maybe a stylistic thing to have extensions for certain file types. What are you using that requires the extension to be able to load a "jsx" file?

**Comment by bruderstein** [comment]
Ah, thank you. I'm using browserify, but hadn't enabled .jsx as an extension, so had to always specify the .jsx in the require.  I'll add the option then it's not required. Sorry for the noise, and thanks for a great plugin - it's a real time saver.

**Comment by jballant** [comment]
No worries. Glad it's useful for you.
